packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -0,0 +1,499 @@
+            );
+        }
+
+        if (!this.props.multiselectable && initialSelection.length > 1) {
+            initialSelection = initialSelection.slice(0, 1);
+        }
+
+        window.addEventListener("click", this.handleWindowClick);
+
+        this.updateSelection(initialSelection);
+        this.toggleMenu(this.checkPropsForMenuState());
+    }
+
+    public componentWillUnmount(): void {
+        window.removeEventListener("click", this.handleWindowClick);
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                {...this.unhandledProps()}
+                aria-disabled={this.props.disabled || null}
+                ref={this.rootElement}
+                className={this.generateClassNames()}
+                onKeyDown={this.handleKeydown}
+                onClick={this.handleClick}
+            >
+                {this.renderContentDisplay()}
+                {this.renderHiddenSelectElement()}
+                {this.renderMenu()}
+            </div>
+        );
+    }
+
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props.managedClasses, "select", "");
+
+        if (this.props.disabled) {
+            className = className.concat(
+                " ",
+                get(this.props.managedClasses, "select__disabled", "")
+            );
+        }
+
+        if (this.state.isMenuOpen) {
+            className = className.concat(
+                " ",
+                get(this.props.managedClasses, "select_menu__open", "")
+            );
+        }
+
+        if (this.props.multiselectable) {
+            className = className.concat(
+                " ",
+                get(this.props.managedClasses, "select__multiSelectable", "")
+            );
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Determine menu state based on props
+     */
+    private checkPropsForMenuState = (): boolean => {
+        let shouldMenuBeOpen: boolean = false;
+
+        if (this.props.isMenuOpen !== undefined) {
+            shouldMenuBeOpen = this.props.isMenuOpen;
+        } else if (this.props.multiselectable === true) {
+            shouldMenuBeOpen = true;
+        }
+
+        return shouldMenuBeOpen;
+    };
+
+    /**
+     * Renders a hidden select element which can interact with a
+     * form hosting this component
+     */
+    private renderHiddenSelectElement(): React.ReactNode {
+        return (
+            <select
+                required={this.props.required || null}
+                name={this.props.name || null}
+                form={this.props.form || null}
+                value={this.state.value}
+                multiple={this.props.multiselectable || null}
+                disabled={this.props.disabled || null}
+                onChange={this.onSelectValueChange}
```

Comment:
It bears asking the question @chrisdholt asked earlier, should the unhandledProps be on the hidden select element and removed from the handled props entirely?

Reply:
Some of these are actually used by the select component code so they are technically "handled".  My general take is that since we're dealing with a finite list of props for native select and this is very much a "component internal" I'd be more worried that arbitrarily passing props down is more likely to cause weird bugs than provide a useful api.

Reply:
This is fine